api.planx.uk/src/app/secure.ts

```diff
@@ -0,0 +1,265 @@
+      header: true,
+    }).pipe(res);
+    res.header("Content-type", "text/csv");
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post("/sign-s3-upload", async (req, res, next) => {
+  if (!req.body.filename) next({ status: 422, message: "missing filename" });
+
+  try {
+    const { fileType, url, acl } = await signS3Upload(req.body.filename);
+
+    res.json({
+      upload_to: url,
+      public_readonly_url_will_be: url.split("?")[0],
+      file_type: fileType,
+      acl,
+    });
+  } catch (err) {
+    next(err);
+  }
+});
+
+const trackAnalyticsLogExit = async (id: number, isUserExit: boolean) => {
+  try {
+    const result = await client.request(
+      `
+      mutation UpdateAnalyticsLogUserExit($id: bigint!, $user_exit: Boolean) {
+        update_analytics_logs_by_pk(
+          pk_columns: {id: $id},
+          _set: {user_exit: $user_exit}
+        ) {
+          id
+          user_exit
+          analytics_id
+        }
+      }
+    `,
+      {
+        id,
+        user_exit: isUserExit,
+      }
+    );
+
+    const analytics_id = result.update_analytics_logs_by_pk.analytics_id;
+    await client.request(
+      `
+      mutation SetAnalyticsEndedDate($id: bigint!, $ended_at: timestamptz) {
+        update_analytics_by_pk(pk_columns: {id: $id}, _set: {ended_at: $ended_at}) {
+          id
+        }
+      }
+    `,
+      {
+        id: analytics_id,
+        ended_at: isUserExit ? new Date().toISOString() : null,
+      }
+    );
+  } catch (e) {
+    // We need to catch this exception here otherwise the exception would become an unhandle rejection which brings down the whole node.js process
+    console.error(
+      "There's been an error while recording metrics for analytics but because this thread is non-blocking we didn't reject the request",
+      (e as Error).stack
+    );
+  }
+
+  return;
+};
+
+app.post("/analytics/log-user-exit", async (req, res, next) => {
+  const analyticsLogId = Number(req.query.analyticsLogId);
+  if (analyticsLogId > 0) trackAnalyticsLogExit(analyticsLogId, true);
+  res.send();
+});
+
+app.post("/analytics/log-user-resume", async (req, res, next) => {
+  const analyticsLogId = Number(req.query.analyticsLogId);
+  if (analyticsLogId > 0) trackAnalyticsLogExit(analyticsLogId, false);
+  res.send();
+});
+
+assert(process.env.GOVUK_NOTIFY_API_KEY);
+app.post(
+  "/send-email/:template",
+  sendEmailLimiter,
+  useSendEmailAuth,
+  sendSaveAndReturnEmail
+);
+app.post("/resume-application", sendEmailLimiter, resumeApplication);
+app.post("/validate-session", validateSession);
+
+assert(process.env.HASURA_PLANX_API_KEY);
+app.use("/webhooks/hasura", useHasuraAuth);
+app.post("/webhooks/hasura/delete-expired-sessions", hardDeleteSessions);
+app.post("/webhooks/hasura/create-reminder-event", createReminderEvent);
+app.post("/webhooks/hasura/create-expiry-event", createExpiryEvent);
+app.post("/webhooks/hasura/send-slack-notification", sendSlackNotification);
+
+export default app;
```

Comment:
thanks for this proposal! i think the only sticky point for me on first read through is also this "cookieless" & "secure" split, which doesn't quite feel like the best fitting nomenclature just yet.

in my mind, the language to describe & group these endpoints is perhaps closer to "authenticated" & "public" ? 

where "authenticated" would group endpoints that use our own auth methods (eg `useJWT`, `useHasuraAuth`) and/or a service-specific token/credential in its request (eg Pay, BOPS & Uniform, etc).

and "public" would group our endpoints that are open, no auth or jwt/cookie check necessary (eg `/gis/`, `/analytics/`, etc). 

does that also make sense to others? 

"cookieless" especially feels like it might get a bit conflated with the client-side concept of cookies. and per our planx service agreements, it's emphasized that NO parts of the public-facing planx service use or leave cookies on applicants machines (eg why we are able to not have a cookies banner)!